compiler/src/dotty/tools/MainGenericRunner.scala

```diff
@@ -102,7 +105,18 @@ object MainGenericRunner {
     case "-run" :: fqName :: tail =>
       process(tail, settings.withExecuteMode(ExecuteMode.Run).withTargetToRun(fqName))
     case ("-cp" | "-classpath" | "--class-path") :: cp :: tail =>
-      process(tail, settings.copy(classPath = settings.classPath.appended(cp)))
+      val (tailargs, cpstr) = if classpathSeparator != ";" || cp.contains(classpathSeparator) then
+        (tail, cp)
+      else
+        // combine globbed classpath entries into a classpath
+        val globdir = cp.replaceAll("[\\/][^\\/]*$","") // must be forward-backward-slash-agnostic
+        val jarfiles = cp :: tail
```

Comment:
Could you check if it is working for `bin/scala -classpath myfile2.jar myfile.jar` (the second jar is with the main class and is supposed to be run). Nonetheless, `bin/scala -classpath myfile2.jar -run myfile.jar` should be completely fine.

Reply:
Wow, cool!  I didn't know you could run a main jar like that (it works for `scala-2.13.6`, but not for `scala3-3.0.2`).
I'll see if we can support it w/out requiring the `-run` arg.

Reply:
I assume this is unrelated to this PR, and it's independent of the platform.   The following are from a Linux session.

A difference between `scala3-3.0.2` and the current release candidate is what happens when trying to run a main class, but it's not found in the classpath.    

Here's a command line, intended to run the main method of object invest.Main, if it's not defined in `myfile2.jar`:

```bash
bin/scala -classpath myfile2.jar invest.Main
```
If I misspell the main object, `scala3-3.02` prints an error message:
```scala
Error: Could not find or load main class invest.Main
Caused by: java.lang.ClassNotFoundException: invest.Main
```

When I attempt the same thing with this release candidate, it starts a REPL session, with no error message:
```scala
Welcome to Scala 3.1.1-RC1-bin-SNAPSHOT-git-b9557b0 (16.0.1, Java OpenJDK 64-Bit Server VM).
Type in expressions for evaluation. Or try :help.
                                                                                                                                            
scala> 
```
If I add `-run` ahead of the main object name, it prints the following error:
```scala
philwalk@d5:/opt/ue$ /c/opt/scala3/bin/scala -run invest.Main
Exception in thread "main" java.nio.file.NoSuchFileException: invest.Main
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:375)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:426)
	at java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:420)
	at java.base/java.nio.file.Files.newInputStream(Files.java:160)
	at dotty.tools.io.File.inputStream(File.scala:51)
	at dotty.tools.io.Jar.withJarInput(Jar.scala:59)
	at dotty.tools.io.Jar.manifest(Jar.scala:47)
	at dotty.tools.io.Jar.mainClass(Jar.scala:49)
	at dotty.tools.MainGenericRunner$.$anonfun$17(MainGenericRunner.scala:195)
	at scala.Option.flatMap(Option.scala:283)
	at dotty.tools.MainGenericRunner$.run$1(MainGenericRunner.scala:201)
	at dotty.tools.MainGenericRunner$.main(MainGenericRunner.scala:220)
	at dotty.tools.MainGenericRunner.main(MainGenericRunner.scala)
```
The message seems to suggest that it interprets `invest.Main` as a file path, rather than a main object.

Reply:
That's interesting, I was working on that 2 weeks ago and I thought I covered it with proper testing. Will look into that one tomorrow morning.

Reply:
Actually, I think this is desired behaviour without `-run` flag, and with `-run` we should catch error and create proper message. The thing is, without `-run` flag we cannot be sure what is the string we are processing - it can be part of some longer flag, like `-mySetting myProperty`. There is heuristic for guessing that string is possibly the main class. If it is not, we assume it's part of residual args. Therefore the REPL start. It's used if somebody want to start REPL with custom configuration.

Reply:
And for `-run` we firstly try to run it as a main class from classpath, and if it fails we try to run it as a possible runnable file, therefore the error is about `NoSuchFile`, because it fails at fallback. We should create proper message like: The string: <value> is not a valid main class fq name from classpath, nor the valid runnable file.

Reply:
I know that starting REPL session can be misleading, but I don't know if we could create an efficient parser for all the cases that are: `run` `script` `compile` `repl` that can infer what is the mode that user was thinking about, so we cover happy cases and hope that user knows what he's doing and he won't provide not existing main class. The settings like `-run` and `-script` should help enforcing mode so I think that error messages are the thing we should improve right now.

Reply:
Okay, sounds good.